src/zenml/stack/utils.py

```diff
@@ -75,7 +75,16 @@ def validate_stack_component_config(
             return None
         raise
 
-    configuration = flavor_class.config_class(**configuration_dict)
+    config_class = flavor_class.config_class
+    # Make sure extras are forbidden for the config class. Due to inheritance
```

Comment:
As you mentioned in the description, for whatever reason, if a user sets the `extra` to `allow`, perhaps we can use a warning or a debug message here.

Reply:
The problem is, I can't even detect that right? Whether someone did it explicitly or whether it happened due to inheritance of `BaseSettings`

Reply:
True. Feel free to skip it.